Installation issue: intel-tbb

### Steps to reproduce the issue

```console
$ spack spec -I intel-tbb
Input spec
--------------------------------
 -   intel-tbb

Concretized
--------------------------------
 -   intel-tbb@2021.7.0%gcc@13.0.1~ipo+shared+tm build_system=cmake build_type=RelWithDebInfo cxxstd=default generator=make arch=linux-fedora38-zen2
[+]      ^cmake@3.26.3%gcc@13.0.1~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-fedora38-zen2
[+]          ^ncurses@6.4%gcc@13.0.1~symlinks+termlib abi=none build_system=autotools arch=linux-fedora38-zen2
[+]          ^openssl@1.1.1t%gcc@13.0.1~docs~shared build_system=generic certs=mozilla arch=linux-fedora38-zen2
[+]              ^ca-certificates-mozilla@2023-01-10%gcc@13.0.1 build_system=generic arch=linux-fedora38-zen2
[+]              ^perl@5.36.0%gcc@13.0.1+cpanm+open+shared+threads build_system=generic arch=linux-fedora38-zen2
[+]                  ^berkeley-db@18.1.40%gcc@13.0.1+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-fedora38-zen2
[+]                  ^bzip2@1.0.8%gcc@13.0.1~debug~pic+shared build_system=generic arch=linux-fedora38-zen2
[+]                      ^diffutils@3.9%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]                  ^gdbm@1.23%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]                      ^readline@8.2%gcc@13.0.1 build_system=autotools patches=bbf97f1 arch=linux-fedora38-zen2
[+]              ^zlib@1.2.13%gcc@13.0.1+optimize+pic+shared build_system=makefile arch=linux-fedora38-zen2
[+]      ^gmake@4.4.1%gcc@13.0.1~guile build_system=autotools arch=linux-fedora38-zen2
[+]      ^hwloc@2.9.1%gcc@13.0.1~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-fedora38-zen2
[+]          ^libpciaccess@0.17%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]              ^libtool@2.4.7%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]                  ^m4@1.4.19%gcc@13.0.1+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-fedora38-zen2
[+]                      ^libsigsegv@2.14%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]              ^util-macros@1.19.3%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
[+]          ^libxml2@2.10.3%gcc@13.0.1~python build_system=autotools arch=linux-fedora38-zen2
[+]              ^libiconv@1.17%gcc@13.0.1 build_system=autotools libs=shared,static arch=linux-fedora38-zen2
[+]              ^xz@5.4.1%gcc@13.0.1~pic build_system=autotools libs=shared,static arch=linux-fedora38-zen2
[+]          ^pkgconf@1.8.0%gcc@13.0.1 build_system=autotools arch=linux-fedora38-zen2
```


### Error message

<details><summary>Error message</summary>

<pre>
==> Installing intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr
==> No binary for intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr found: installing from source
==> Using cached archive: /opt/spack/var/spack/cache/_source-cache/archive/2c/2cae2a80cda7d45dc7c072e4295c675fff5ad8316691f26f40539f7e7e54c0cc.tar.gz
==> No patches needed for intel-tbb
==> intel-tbb: Executing phase: 'cmake'
==> intel-tbb: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/opt/spack/opt/spack/linux-fedora38-zen2/gcc-13.0.1/gmake-4.4.1-zpfldlcucu24lmll33jzwwdj7xpx73ss/bin/make' '-j16'

4 errors found in build log:
     102    cd /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-build-akcv2vx/src/tbb && /
            opt/spack/lib/spack/env/gcc/g++ -D__TBB_BUILD -D__TBB_USE_ITT_NOTIFY -I/tmp/root/spack-stage/spack-stage-intel-tbb-2021.7
            .0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/src/tbb/../../include -O2 -g -DNDEBUG -std=c++11 -flto=auto -fno-fat-lto-ob
            jects -fPIC -flifetime-dse=1 -Wall -Wextra -Wfatal-errors -mrtm -mwaitpkg -MD -MT src/tbb/CMakeFiles/tbb.dir/concurrent_b
            ounded_queue.cpp.o -MF CMakeFiles/tbb.dir/concurrent_bounded_queue.cpp.o.d -o CMakeFiles/tbb.dir/concurrent_bounded_queue
            .cpp.o -c /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/src/tbb/concurr
            ent_bounded_queue.cpp
     103    cd /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-build-akcv2vx/src/tbbmallo
            c && /opt/spack/lib/spack/env/gcc/g++ -D__TBBMALLOC_BUILD -D__TBB_USE_ITT_NOTIFY -I/tmp/root/spack-stage/spack-stage-inte
            l-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/src/tbbmalloc/../../include -O2 -g -DNDEBUG -std=c++11 -flto=au
            to -fno-fat-lto-objects -fPIC -flifetime-dse=1 -fno-rtti -fno-exceptions -mrtm -mwaitpkg -MD -MT src/tbbmalloc/CMakeFiles
            /tbbmalloc.dir/__/tbb/itt_notify.cpp.o -MF CMakeFiles/tbbmalloc.dir/__/tbb/itt_notify.cpp.o.d -o CMakeFiles/tbbmalloc.dir
            /__/tbb/itt_notify.cpp.o -c /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-s
            rc/src/tbb/itt_notify.cpp
     104    cd /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-build-akcv2vx/src/tbb && /
            opt/spack/lib/spack/env/gcc/g++ -D__TBB_BUILD -D__TBB_USE_ITT_NOTIFY -I/tmp/root/spack-stage/spack-stage-intel-tbb-2021.7
            .0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/src/tbb/../../include -O2 -g -DNDEBUG -std=c++11 -flto=auto -fno-fat-lto-ob
            jects -fPIC -flifetime-dse=1 -Wall -Wextra -Wfatal-errors -mrtm -mwaitpkg -MD -MT src/tbb/CMakeFiles/tbb.dir/arena_slot.c
            pp.o -MF CMakeFiles/tbb.dir/arena_slot.cpp.o.d -o CMakeFiles/tbb.dir/arena_slot.cpp.o -c /tmp/root/spack-stage/spack-stag
            e-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/src/tbb/arena_slot.cpp
     105    cd /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-build-akcv2vx/src/tbb && /
            opt/spack/lib/spack/env/gcc/g++ -D__TBB_BUILD -D__TBB_USE_ITT_NOTIFY -I/tmp/root/spack-stage/spack-stage-intel-tbb-2021.7
            .0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/src/tbb/../../include -O2 -g -DNDEBUG -std=c++11 -flto=auto -fno-fat-lto-ob
            jects -fPIC -flifetime-dse=1 -Wall -Wextra -Wfatal-errors -mrtm -mwaitpkg -MD -MT src/tbb/CMakeFiles/tbb.dir/dynamic_link
            .cpp.o -MF CMakeFiles/tbb.dir/dynamic_link.cpp.o.d -o CMakeFiles/tbb.dir/dynamic_link.cpp.o -c /tmp/root/spack-stage/spac
            k-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/src/tbb/dynamic_link.cpp
     106    In file included from /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/tes
            t/tbbmalloc/test_malloc_lib_unload.cpp:21:
     107    /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/test/common/utils_assert.
            h: In function 'void utils::ReportError(const char*, int, const char*, const char*)':
  >> 108    /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/test/common/utils_assert.
            h:37:5: error: 'abort' was not declared in this scope
     109       37 |     abort();
     110          |     ^~~~~
     111    /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/test/common/utils_assert.
            h:22:1: note: 'abort' is defined in header '<cstdlib>'; did you forget to '#include <cstdlib>'?
     112       21 | #include "utils_report.h"
     113      +++ |+#include <cstdlib>
     114       22 |
  >> 115    make[2]: *** [test/CMakeFiles/_test_malloc_lib_unload.dir/build.make:79: test/CMakeFiles/_test_malloc_lib_unload.dir/tbbm
            alloc/test_malloc_lib_unload.cpp.o] Error 1
     116    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-b
            uild-akcv2vx'
  >> 117    make[1]: *** [CMakeFiles/Makefile2:4038: test/CMakeFiles/_test_malloc_lib_unload.dir/all] Error 2
     118    make[1]: *** Waiting for unfinished jobs....
     119    [  7%] Building CXX object src/tbb/CMakeFiles/tbb.dir/exception.cpp.o
     120    cd /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-build-akcv2vx/src/tbb && /
            opt/spack/lib/spack/env/gcc/g++ -D__TBB_BUILD -D__TBB_USE_ITT_NOTIFY -I/tmp/root/spack-stage/spack-stage-intel-tbb-2021.7
            .0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/src/tbb/../../include -O2 -g -DNDEBUG -std=c++11 -flto=auto -fno-fat-lto-ob
            jects -fPIC -flifetime-dse=1 -Wall -Wextra -Wfatal-errors -mrtm -mwaitpkg -MD -MT src/tbb/CMakeFiles/tbb.dir/exception.cp
            p.o -MF CMakeFiles/tbb.dir/exception.cpp.o.d -o CMakeFiles/tbb.dir/exception.cpp.o -c /tmp/root/spack-stage/spack-stage-i
            ntel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/src/tbb/exception.cpp
     121    [  7%] Building CXX object src/tbb/CMakeFiles/tbb.dir/governor.cpp.o
     122    cd /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-build-akcv2vx/src/tbb && /
            opt/spack/lib/spack/env/gcc/g++ -D__TBB_BUILD -D__TBB_USE_ITT_NOTIFY -I/tmp/root/spack-stage/spack-stage-intel-tbb-2021.7
            .0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/src/tbb/../../include -O2 -g -DNDEBUG -std=c++11 -flto=auto -fno-fat-lto-ob
            jects -fPIC -flifetime-dse=1 -Wall -Wextra -Wfatal-errors -mrtm -mwaitpkg -MD -MT src/tbb/CMakeFiles/tbb.dir/governor.cpp
            .o -MF CMakeFiles/tbb.dir/governor.cpp.o.d -o CMakeFiles/tbb.dir/governor.cpp.o -c /tmp/root/spack-stage/spack-stage-inte
            l-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-src/src/tbb/governor.cpp
     123    [  7%] Building CXX object src/tbb/CMakeFiles/tbb.dir/global_control.cpp.o

     ...

     187    [ 13%] Linking CXX executable ../gnu_13.0_cxx11_64_relwithdebinfo/test_malloc_whitebox
     188    cd /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-build-akcv2vx/test && /opt
            /spack/opt/spack/linux-fedora38-zen2/gcc-13.0.1/cmake-3.26.3-z5wrsf2ol5sx2b2rl3wxqgoctvn7obwc/bin/cmake -E cmake_link_scr
            ipt CMakeFiles/test_malloc_whitebox.dir/link.txt --verbose=1
     189    /opt/spack/lib/spack/env/gcc/g++ -O2 -g -DNDEBUG -rdynamic CMakeFiles/test_malloc_whitebox.dir/tbbmalloc/test_malloc_whit
            ebox.cpp.o -o ../gnu_13.0_cxx11_64_relwithdebinfo/test_malloc_whitebox  -ldl
     190    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-b
            uild-akcv2vx'
     191    [ 13%] Built target test_malloc_whitebox
     192    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-b
            uild-akcv2vx'
  >> 193    make: *** [Makefile:169: all] Error 2

See build log for details:
  /tmp/root/spack-stage/spack-stage-intel-tbb-2021.7.0-akcv2vxhz44btb4ldywb7jpowmwfvatr/spack-build-out.txt
</pre></details>


### Information on your system

* **Spack:** 0.20.0.dev0 (4363b1c2dcdae9245054cd44124f8ac6789bfd39)
* **Python:** 3.11.3
* **Platform:** linux-fedora38-zen2
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/11356321/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/11356322/spack-build-env.txt)

@rscohn2

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/intel-tbb/gcc_13-2021.patch
+++ b/var/spack/repos/builtin/packages/intel-tbb/gcc_13-2021.patch
@@ -1,0 +1,27 @@
+From b66c3b217f1954e31ea6f40a2416dd3264861c66 Mon Sep 17 00:00:00 2001
+From: Sam James <sam@gentoo.org>
+Date: Tue, 21 Feb 2023 09:36:33 +0000
+Subject: [PATCH] test: common: include <cstdlib> for abort() (fix build with
+ GCC 13)
+
+GCC 13 (as usual for new compiler releases) shuffles around some
+internal includes and so <cstdint> etc is no longer transitively included.
+
+See https://www.gnu.org/software/gcc/gcc-13/porting_to.html.
+
+Signed-off-by: Sam James <sam@gentoo.org>
+---
+ test/common/utils_assert.h | 4 +++-
+ 1 file changed, 3 insertions(+), 1 deletion(-)
+
+diff --git a/test/common/utils_assert.h b/test/common/utils_assert.h
+index 1df8ae72acc49fe38dac4d9bed4e9f4f26affcf5..0123ab881e124a800a5ebf8507050148038747d5 100644
+@@ -20,6 +20,8 @@
+ #include "config.h"
+ #include "utils_report.h"
+ 
++#include <cstdlib>
++
+ #define REPORT_FATAL_ERROR REPORT
+ 
+ namespace utils {

--- a/var/spack/repos/builtin/packages/intel-tbb/package.py
+++ b/var/spack/repos/builtin/packages/intel-tbb/package.py
@@ -122,6 +122,10 @@ class IntelTbb(CMakePackage, MakefilePackage):
     patch("gcc_generic-pedantic-2019.patch", level=1, when="@2019.1:2019.5")
     patch("gcc_generic-pedantic-4.4.patch", level=1, when="@:2019.0")
 
+    # Patch and conflicts for GCC 13 support (#1031).
+    patch("gcc_13-2021.patch", when="@2021.1:")
+    conflicts("%gcc@13", when="@:2021.3")
+
     # Patch cmakeConfig.cmake.in to find the libraries where we install them.
     patch("tbb_cmakeConfig-2019.5.patch", level=0, when="@2019.5:2021.0")
     patch("tbb_cmakeConfig.patch", level=0, when="@2017.7:2019.4")

